packages/extras/src/lib/components/Gizmo/Gizmo.svelte

```diff
@@ -1,467 +1,143 @@
-  const finalQuaternion = new Quaternion()
-  let radius = 0
-
-  let animating = false
-  const mouse = new Vector2()
-  const raycaster = new Raycaster()
-
-  /**
-   * Floating point operations make it hard to compare quaternions, controls
-   * (such as the OrbitControls) may also restrict the rotation of the camera on
-   * certain axes. To allow for loose equality checks, we use a sensible
-   * threshold to compare quaternions.
-   *
-   * @param a - Quaternion a
-   * @param b - Quaternion b
-   * @param threshold - Threshold to use for comparison
-   */
-  const quaternionsAreEqual = (a: Quaternion, b: Quaternion, threshold: number) => {
-    const delta =
-      Math.abs(a.x - b.x) + Math.abs(a.y - b.y) + Math.abs(a.z - b.z) + Math.abs(a.w - b.w)
-    return delta < threshold
-  }
-
-  /**
-   * @returns boolean that indicates if the target and the current rotation are equal.
-   */
-  const handleIntersection = (intersection: Intersection<Object3D>): boolean => {
-    const object = intersection.object
-    const targetPos = object.userData.targetPosition as [number, number, number]
-    const targetEuler = object.userData.targetEuler as [number, number, number]
-
-    radius = camera.current.position.distanceTo(centerVec)
-    targetPosition
-      .set(...targetPos)
-      .multiplyScalar(radius)
-      .add(centerVec)
-    targetQuaternion.setFromEuler(new Euler(...targetEuler))
-
-    const dummy = new Object3D()
-    dummy.position.copy(centerVec)
-
-    dummy.lookAt(camera.current.position)
-    currentQuaternion.copy(dummy.quaternion)
-
-    dummy.lookAt(targetPosition)
-    finalQuaternion.copy(dummy.quaternion)
-
-    if (quaternionsAreEqual(finalQuaternion, currentQuaternion, 0.0001)) {
-      return true
-    }
-
-    animating = true
-    return false
-  }
-
-  const handleClick = (event: MouseEvent) => {
-    if (animating) {
-      return
-    }
-
-    // Raycasting is done manually.
-    const rect = clickTarget.getBoundingClientRect()
-    const offsetX = rect.left + (clickTarget.offsetWidth - size)
-    const offsetY = rect.top + (clickTarget.offsetHeight - size)
-    mouse.x = ((event.clientX - offsetX) / (rect.right - offsetX)) * 2 - 1
-    mouse.y = -((event.clientY - offsetY) / (rect.bottom - offsetY)) * 2 + 1
-
-    raycaster.setFromCamera(mouse, orthoCam)
-
-    // Gaurd against undefined axis sprites otherwise TS will complain
-    if (!posX || !posY || !posZ || !negX || !negY || !negZ) {
-      return
-    }
-
-    const intersects = raycaster.intersectObjects([posX, posY, posZ, negX, negY, negZ])
+  $effect(() => {
+    if (!_controls) return
 
-    if (intersects.length > 0) {
-      const alreadyReached = handleIntersection(intersects[0])
-      if (alreadyReached) {
-        // get the second closest intersection
-        if (intersects.length > 1) {
-          handleIntersection(intersects[1])
-        }
+    const handleStart = () => (_controls.enabled = false)
```

Comment:
it looks like attaching the gizmo will do this already i think https://github.com/Fennec-hub/three-viewport-gizmo/blob/291992c3a03648acf9ac7318a9c38ad7e4c797d6/lib/ViewportGizmo.ts#L377-L381

Reply:
good catch, fixed up!